### ISSUE TITLE
"The current project has been modified." -- not

### ISSUE THREAD

**Issue by phil-pilgrim** [parallaxinc/solo#405]
This message pops up whenever I want to load a different program or create a new one, whether I've made any changes or not. Does Solo use a "dirty" bit that gets set whenever a change is made, cleared when the project is saved, and checked when the user wants to exit a project? Or is this just a blanket warning that's issued on purpose regardless of the project's change status?

I'm afraid that a system that cries "Wolf!" will eventually be ignored.

**Comment by zfi** [comment]
Solo makes a copy of the project, the initial state, as it is either loaded from a file or created new. It then compares this copy of the project with the current state of the project to determine if anything has changed. This includes some subtle details such as block position changes or changes in any blocks enable state. The initial state is updated when the project is saved and this becomes the new initial state.

Given how this is supposed to work, are you seeing an alternate behavior? And if you are, could you list the steps to reproduce the error?

**Comment by phil-pilgrim** [comment]
> Given how this is supposed to work, are you seeing an alternate behavior? And if you are, could you list the steps to reproduce the error?

Yes I am. I can load a project and, without changing anything, attempt to load another project right away, and get this message. Basically, I _always_ get this message when I try to exit a project.

**Comment by zfi** [comment]
Can you verify that you are seeing this on solo.parallax.com? I am able to load a project, click on the Open button and load a different project without seeing the project changed message.

**Comment by phil-pilgrim** [comment]
No. I'm seeing it here:

https://phil-pilgrim.github.io/solo/blocklyc.html

It's a derivative of Matt's Solo site.

**Comment by zfi** [comment]
In researching this issue, I did find a related issue that appears to be triggered by specific blocks. In this issue, the Blockly core is returning project code that is not identical to the project code that was included when the project is created in Blockly. In simpler terms, there appear to be blocks that will trigger the 'project changed' code immediately.  I will create new issue for that.

As to the present issue, the problem reported isn't about the code in this repository. I tried filing an issue on @MatzElectronics fork but issues are disabled.

**Comment by zfi** [comment]
Update: You can file an issue [here](https://github.com/MatzElectronics/solo/issues).

**Comment by zfi** [comment]
The code that the Blockly core was returning was indeed different but the cause was unexpected. 

<img width="393" alt="Screen Shot 2020-04-03 at 11 18 57 PM" src="https://user-images.githubusercontent.com/8442260/78420246-376b1c00-7602-11ea-90d1-86345cecd2a7.png">

The project contained an older version of the LIS3DH initialization. When it was loaded, the init block parameters that had changed were set to a default value. This caused the "project has changed" code to identify the change and raise the dialog.

**Comment by phil-pilgrim** [comment]
Thanks, Jim. I posted on Matt's Solo page.

**Comment by zfi** [comment]
Closing this issue.
